webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/index.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import {Button} from 'react-bootstrap';
+import Disk from './disk';
+import Checkbox from '../../../../common/forms/Checkbox';
+import './controller.scss';
+
+const Controller = ({
+  addDiskEnabled,
+  addDisk,
+  removeDisk,
+  updateController,
+  updateDisk,
+  ControllerTypes,
+  controller
+}) => {
+  const _updateController = (attribute, e) => {
+    const value = e.target.type === 'checkbox' ? e.target.checked : e.target.value;
+
+    updateController({[attribute]: value});
+  };
+
+  const _updateDisk = (diskIndex, attribute, e) => {
+    const value = e.target.type === 'checkbox' ?
+      e.target.checked :
+      e.target.value;
+
+    updateDisk(diskIndex, {[attribute]: value});
+  };
+
+  const selectableTypes = () => {
+    return Object
+      .entries(ControllerTypes)
+      .map(attribute => {
+        return (
+          <option key={attribute[0]} value={attribute[0]}>{attribute[1]}</option>
+        );
+      });
+  };
+
+  const disks = () => {
+    return controller
+      .disks
+      .map((disk, index) => {
+        return (<Disk
+          key={index}
+          id={index}
+          updateDisk={_updateDisk.bind(this, index)}
+          removeDisk={removeDisk.bind(this, index)}
+          datastores={controller.datastores}
+          storagePods={controller.storage_pods}
+          {...disk}/>);
+      });
+  };
+
+  return (
+    <div className="controller-container">
+      <div className="controller-header">
+        <div className="control-label col-md-2 controller-selected-container">
+          <Checkbox
+            className="cb-controller-is-selected"
+            checked={controller.isSelected}
+            onChange={_updateController.bind(this, 'isSelected')}/>
+          <label>{__('Create SCSI controller')}</label>
```

Comment:
I think the design is flawed then. It would be better to have a dedicated delete button per controller (we already have a dedicated button per disk). The checkbox is very confusing. I didn't get, that it's related to the `Delete Controller` button.

![image](https://cloud.githubusercontent.com/assets/4107560/23789283/73b5a410-057a-11e7-9f5f-20b8353e4d73.png)

btw: This is how this looks in VMWare. The UI is pretty bad I think, but Foreman users might be familiar with this:

![image](https://cloud.githubusercontent.com/assets/4107560/23789381/ed0bc39e-057a-11e7-81b8-6e56269f01ab.png)
(Sorry for the German language, but I guess you get the point)

@Rohoover: Do you agree? Any more comments?